core/src/main/java/org/infinispan/Cache.java

```diff
@@ -357,4 +361,40 @@
    default void shutdown() {
       stop();
    }
+
```

Comment:
Please add documentation for the methods inherited from the `Map` prohibiting the user from modifying the provided value; the user **must** create a copy or completely new instance. This could be enforced in the future if that's technically possible.